File: Data Structures & Algorithms/plus-one/submission-1.cpp
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        int ten = 10;
        long long res = digits[digits.size()-1];
        for(int i = digits.size()-2; i>=0; --i){ // transfer this to a number
            res += ten * digits[i];  // tenth addition
            ten = 10 * ten; // next iteration
        }
        res ++; 
        vector<int> ans;
        while(res!=0){
            int digit = res %10;
            res /= 10;
            ans.push_back(digit);

        }
        reverse(ans.begin(), ans.end());
        return ans;
    }
};

File: Data Structures & Algorithms/palindromic-substrings/submission-1.py
class Solution:
    def countSubstrings(self, s: str) -> int:
        res = 0
        for i in range(len(s)):
            l, r = i, i
            def ispali(l, r):
                res = 0
                while l >= 0 and r <len(s) and s[l] == s[r]:
                    res+=1
                    l-=1
                    r+=1
                return res
            res += ispali(l, r)
            l, r = i, i+1
            res+=ispali(l, r)
        return res